client/lib/media/utils.js

```diff
@@ -232,6 +233,28 @@ var MediaUtils = {
 	},
 
 	/**
+	 * Returns true if the specified item is a valid file in a Premium plan,
+	 * or false otherwise.
+	 *
+	 * @param  {Object}  item Media object
+	 * @param  {Object}  site Site object
+	 * @return {Boolean}      Whether the Premium plan supports the item
+	 */
+	isSupportedFileTypeInPremium: function( item, site ) {
+		if ( ! site || ! item ) {
+			return false;
```

Comment:
would it be more accurate here to return undefined, since we don't know yet whether the file is supported?